API/contacts/SearchContact.php

```diff
@@ -0,0 +1,37 @@
+<?php
+    ini_set('display_errors', 1);
+    ini_set('error_reporting', E_ALL);
+
+    require_once __DIR__ . '/../utils/JsonUtils.php';
+    require_once __DIR__ . '/../utils/ResponseSender.php';
+    require_once __DIR__ . '/../utils/RequestReciever.php';
+    require_once __DIR__ . '/../utils/ResponseCodes.php';
+
+    require_once __DIR__ . '/Contact.php';
+    require_once __DIR__ . '/ContactAPI.php';
+
+    require_once __DIR__ . '/../database/Database.php';
+
+    $contact = RequestReciever::recieveGET(new Contact());
+    
+    $database = new Database();
+
+    $mysql = $database->connectToDatabase();
+
+    if ($contact == false)
+        ResponseSender::send(ResponseCodes::NOT_FOUND, "Missing request body");
+
+    $query = !empty($contact->firstName) ? $contact->firstName : "";
+    $query = $query . (!empty($contact->lastName) ? " " . $contact->lastName : ""); 
+    $query = $query . (!empty($contact->phone) ? " " . $contact->phone : "");
+    $query = $query . (!empty($contact->email) ? " " . $contact->email : "");
+
+    $contactAPI = new ContactAPI($mysql);
+
+    $result = $contactAPI->GetContact($query, $contact->userID, 10);
```

Comment:
The value for number of results should be a parameter rather than hard-coded. Is this something you would want us to pass from the front end when searching?

Reply:
Not sure about this part yet... I need to think about how I would handle it backend wise... I will think about later, for now it will be hardcoded

Reply:
I came up with an idea how I can make it work... I will implement when I will be free.

Reply:
Can you just add a TODO for now and make a github issue for taking care of it? That way we can merge the rest